src/Platform.Handlers/src/Xamarin.Platform.Handlers/Handlers/View/AbstractViewHandler.cs

```diff
@@ -43,16 +43,28 @@ protected AbstractViewHandler(PropertyMapper mapper)
 
 		public object? NativeView => TypedNativeView;
 
-		public virtual void SetView(IView view)
+
+		public void SetView(IView view)
 		{
 			_ = view ?? throw new ArgumentNullException(nameof(view));
 
+			if (VirtualView != null)
+			{
+				TearDownOldVirtualView(VirtualView);
```

Comment:
Added `TearDownOldVirtualView` for when the view is replaced with a new virtual view so there is a hook point to do what you want to do

Reply:
We have a `TeardownNativeView`, should be just `TearDownVirtualView`?. Would it make sense to only have a single Teardown method with VirtualView and NativeView as parameters? I am concerned that it could cause some confusion  and not knowing the difference and when to use.

Reply:
The reason for the separation is that the handler is reusable across different VirtualViews. Think CollectionView reusable cells. So, the handler will get different VirtualViews during its life time but it will only ever have the one NativeView. Once the NativeView is torn down then the handler should be considered done with.